src/main/kotlin/org/wfanet/measurement/measurementconsumer/stats/FrequencyMeasurementStatistics.kt

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2023 The Cross-Media Measurement Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wfanet.measurement.measurementconsumer.stats
+
+interface FrequencyMeasurementStatistics {
+
+  val reachMeasurementStatistics: ReachMeasurementStatistics
+
+  /** Outputs [FrequencyVariances] of the given reach-frequency measurement. */
+  fun variances(
+    totalReach: Int,
+    relativeFrequencyDistribution: Map<Int, Double>,
+    params: FrequencyMeasurementParams
+  ): FrequencyVariances {
+    var suffixSum = 0.0
+
+    // There is no estimate of zero-frequency reach
+    val kPlusRelativeFrequencyDistribution: Map<Int, Double> =
+      (params.maximumFrequency downTo 1).associateWith { frequency ->
+        suffixSum += relativeFrequencyDistribution.getOrDefault(frequency, 0.0)
+        suffixSum
+      }
+
+    val reachVariance =
+      reachMeasurementStatistics.variance(
+        totalReach,
+        ReachMeasurementParams(params.vidSamplingIntervalWidth, params.reachDpParams)
+      )
+
+    val relativeVariances: Map<Int, Double> =
```

Comment:
nit: `relativeVariance` or `ratioVariance`? since you used `reachRatio` as argument name below